util/settings/settings.go

```diff
@@ -448,6 +466,31 @@ func (mgr *SettingsManager) GetResourceOverrides() (map[string]v1alpha1.Resource
 		}
 	}
 
+	var diffOptions ArgoCDDiffOptions
+	if value, ok := argoCDCM.Data[resourceCompareOptionsKey]; ok {
+		err := yaml.Unmarshal([]byte(value), &diffOptions)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	ignoreStatus := v1alpha1.ResourceOverride{IgnoreDifferences: "jsonPointers:\n- /status"}
+	crdGK := "apiextensions.k8s.io/CustomResourceDefinition"
+
+	switch diffOptions.IgnoreResourceStatusField {
+	case "", "crd":
```

Comment:
The user might configure some other overrides for CRDs/all resources. Please add a check if override already exists and inject Ignore difference.

Unfortunately  `ResourceOverride` fields `IgnoreDifferences` and `Actions` are strings, instead of corresponding data structure, so we have to mess with YAML.

I suggest to fix it for `IgnoreDifferences`. The simplest way is to create custom JSON marshaller that transforms data structures to strings and string to data structures:

```golang
# move from https://github.com/argoproj/argo-cd/blob/master/util/argo/normalizers/diff_normalizer.go#L27
type OverrideIgnoreDiff struct {
	JSONPointers []string `yaml:"jsonPointers"`
}

type rawResourceOverride struct {
	HealthLua         string           `json:"health.lua,omitempty"`
	Actions           string           `json:"actions,omitempty"`
	IgnoreDifferences string           `json:"ignoreDifferences,omitempty"`
	KnownTypeFields   []KnownTypeField `json:"knownTypeFields,omitempty"`
}

// ResourceOverride holds configuration to customize resource diffing and health assessment
type ResourceOverride struct {
	HealthLua         string             `protobuf:"bytes,1,opt,name=healthLua"`
	Actions           string             `protobuf:"bytes,3,opt,name=actions"`
	IgnoreDifferences OverrideIgnoreDiff `protobuf:"bytes,2,opt,name=ignoreDifferences"`
	KnownTypeFields   []KnownTypeField   `protobuf:"bytes,4,opt,name=knownTypeFields"`
}

func (s *ResourceOverride) UnmarshalJSON(data []byte) error {
	raw := &rawResourceOverride{}
	if err := json.Unmarshal(data, &raw); err != nil {
		return err
	}
	s.KnownTypeFields = raw.KnownTypeFields
	s.HealthLua = raw.HealthLua
	s.Actions = raw.Actions
	return json.Unmarshal([]byte(raw.IgnoreDifferences), &s.IgnoreDifferences)
}

func (s *ResourceOverride) MarshalJSON() ([]byte, error) {
	ignoreDifferencesData, err := json.Marshal(s.IgnoreDifferences)
	if err != nil {
		return nil, err
	}
	raw := &rawResourceOverride{s.HealthLua, s.Actions, string(ignoreDifferencesData), s.KnownTypeFields}
	return json.Marshal(raw)
}
```

What do you think?